apps/src/p5lab/drawUtils.js

```diff
@@ -51,54 +51,86 @@ export function getTextWidth(p5, text, size) {
  */
 export function speechBubble(
   p5,
-  x,
-  y,
-  width,
-  height,
+  bubbleX,
+  bubbleY,
+  bubbleWidth,
+  bubbleHeight,
+  spriteX,
+  spriteY,
   {
-    tailSize = 10,
-    tailTipX = x,
+    tailWidth = 10,
+    tailHeight = 10,
+    tailTipX = spriteX,
     radius = 8,
     fill = 'white',
     strokeWeight = 2,
     stroke = 'gray'
   } = {},
   bubbleType
 ) {
-  const minX = x - width / 2;
-  const minY = y - height - tailSize;
-  const maxY = y - tailSize;
+  // A speech bubble is comprised of a rectangular bubble stacked on top of a bubble.
+  // Bubbles follow the sprite but not past the app canvas. Tails extend down towards the sprite.
+  // The bubble has a fixed size, while the tail can shrink depending on the available space.
+  // bubbleX and bubbleY are the coordinates of the top left corner of the bubble.
```

Comment:
Thanks for adding this helpful info! I think this might actually belong in the jsdoc right above this function. 

for example:

```  // A speech bubble is comprised of a rectangular bubble stacked on top of a bubble.
  // Bubbles follow the sprite but not past the app canvas. Tails extend down towards the sprite.
  // The bubble has a fixed size, while the tail can shrink depending on the available space.
 ```

can probably be added between lines 37 and 39. and then:
// bubbleX and bubbleY are the coordinates of the top left corner of the bubble.


could replace:

```
 * @param {Number} bubbleX - top left corner
 * @param {Number} bubbleY - top left corner 
 ```

more info on jsdocs here: https://jsdoc.app/about-getting-started.html

Reply:
Thanks for this! I've seen this convention used in other places before but didn't actually know what it was called. I'll take a go at modifying.